src/helm/proxy/clients/ice_tokenizer_client.py

```diff
@@ -32,14 +32,9 @@ def tokenize(self, request: TokenizationRequest) -> TokenizationRequestResult:
         try:
 
             def do_it():
-                # TODO: check if tokenize can be replaced with something similar to what we do
-                #  in yalm_tokenizer_client.py:
-                # token_ids = tokenizer.encode(request.text)
-                # tokens = [tokenizer.convert_tokens_to_string([i]) for i in tokenizer.convert_ids_to_tokens(token_ids)]
-                # if request.encode:
-                #     tokens = token_ids
-                # This way we could address #1516 and remove the ▁ in the tests.
                 tokens = tokenizer.encode(request.text) if request.encode else tokenizer.tokenize(request.text)
+                if not request.encode:
+                    tokens = [t.replace("▁", " ") for t in tokens]
```

Comment:
Could we abstract this logic into a separate utility function so it doesn't get duplicated across all the tokenizers?  Is it just the _ issue or are there potentially other things we need to be careful of?

Reply:
It is now in `Client`. And normally it's just the underscore character. previously we had the Gdot character also, but normally this is now handled everywhere. I still added it to the function just in case